unlock-app/src/middlewares/providerMiddleware.ts

```diff
@@ -0,0 +1,51 @@
+import { SET_PROVIDER } from '../actions/provider'
+import { setError } from '../actions/error'
+import { FATAL_MISSING_PROVIDER } from '../errors'
+
+interface Action {
+  type: string
+  [key: string]: any
+}
+
+async function initializeProvider(
+  providerName: string,
```

Comment:
Is there a back story around us passing a string for a provider, then looking up in a dictionary of providers? As the code reads now it looks like we should be able to pass the provide to the initialization method directly

Reply:
We could do that, I think that would be a readability win.

Reply:
the back story is that the provider in redux state must be serializable

Reply:
Right, but we can get the provider one level up and pass it directly into `initializeProvider` instead of passing a bunch of arguments to retrieve within the function

Reply:
Such as in the change I just pushed